sklearn/ensemble/_hist_gradient_boosting/gradient_boosting.py

```diff
@@ -1227,16 +1229,26 @@ class HistGradientBoostingRegressor(RegressorMixin, BaseHistGradientBoosting):
 
         .. versionadded:: 0.24
 
-    monotonic_cst : array-like of int of shape (n_features), default=None
-        Indicates the monotonic constraint to enforce on each feature.
-          - 1: monotonic increase
-          - 0: no constraint
-          - -1: monotonic decrease
+    monotonic_cst : array-like of int of shape (n_features) or dict, default=None
+        Monotonic constraint to enforce on each feature are specified using the
+        following integer values:
 
+        - 1: monotonic increase
+        - 0: no constraint
+        - -1: monotonic decrease
+
+        If a dict with str keys, map feature to monotonic constraints by name.
+        If an array, the feature are mapped to constraints by position.
```

Comment:
What do you think about linking to the subsection in the example from this RP that uses a dictionary for monotonic constraints?

Reply:
How would you do so? with a restructured text reference anchor in the "markdown" cell just before the final code snippet?

Reply:
Done in 1925f0b.